controllers/k8ssandra/reaper.go

```diff
@@ -290,3 +314,69 @@ func (r *K8ssandraClusterReconciler) addClusterToExternalReaper(
 	}
 	return result.Continue()
 }
+
+func addManagementApiSecretsToReaper(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	actualDc *cassdcapi.CassandraDatacenter,
+	logger logr.Logger,
+	authConfig *cassdcapi.ManagementApiAuthManualConfig,
+) error {
+
+	reaperName := kc.Spec.Reaper.ReaperRef.Name
+
+	reaperNamespace := kc.Spec.Reaper.ReaperRef.Namespace
+	if reaperNamespace == "" {
+		reaperNamespace = kc.Namespace
+	}
+
+	tssName := reaper.GetTruststoresSecretName(reaperName)
+	tssKey := client.ObjectKey{Namespace: reaperNamespace, Name: tssName}
+
+	tss := &corev1.Secret{}
+	if err := remoteClient.Get(ctx, tssKey, tss); err != nil {
+		logger.Error(err, "failed to get Reaper's truststore secret")
+		return err
+	}
+
+	cs := &corev1.Secret{}
+	csName := authConfig.ClientSecretName + "-ks"
+	csKey := client.ObjectKey{Namespace: kc.Namespace, Name: csName}
+	if err := remoteClient.Get(ctx, csKey, cs); err != nil {
+		logger.Error(err, "failed to get k8ssandra cluster client secret", "secretName", csName)
+		return err
+	}
+
+	clusterName := cassdcapi.CleanupForKubernetes(actualDc.Spec.ClusterName)
```

Comment:
Issue: I'm not sure this is the right logic - in particular clusterName is not always populated. Is there a function you can re-use in cass operator to figure out the resource names?

Reply:
The `actualDc.Spec.ClusterName` is a field of `CassandraDatacenterSepc` in cass-operator. It's declared like this:
```
	// +kubebuilder:validation:MinLength=2
	ClusterName string `json:"clusterName"`
```
This looks like the field must be a string of at least two characters. Otherwise, I think, the cass-operator's webhook will prevent creating the DC. In other words, if a `CassandraDatacenter` object shows up, this field must be at least 2 chars long. Please correct me if I'm missing something.

Anyhow, we could also take the K8ssandraCluster object name, or `kc.Spec.Cassandra.ClusterName`. I'd say all these should be set to the same thing, but I don't really know.

Could someone else please chime in?

Reply:
Re-reading this... You're only using it for setting names for the truststore/keystore files that get materialised in the secret right? So this will just affect the keys in the aggregated secret that contains all of the encryption materials for the different Cassandra clusters? 

It may not matter to much if that's right, although I'd still suggest you use something like the namespace-name of the k8ssandra cluster since otherwise multiple clusters with the same name in different namespaces can overwrite each other...

Reply:
This is correct. I'll make a fix.

Reply:
After some investigation, and a chat with Alex, we concluded that this is, in fact, a blocking issue.
We need a proper way of making Reaper (and its truststore) capable to tell two equally named clusters from different namespaces apart. It's not just for encryption, but for monitoring also.

We'll be putting this ticket on hold until we fix this in Reaper.